### DISCUSSION TITLE
Gists Don't Have Search or Categories/Lists

### DISCUSSION THREAD

**Discussion by Justintime50** [community/community#38820] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Gists are a great place to store snippets or short guides while also being a terrible place to navigate. There is no way to search through my various pages of Gists nor is there a way to organize them into lists or categories like repos can. This was initially brought up at [GitHub Universe](https://github.com/community/events/discussions/44) where I would have hoped they'd make some announcement about Gists getting some love.

Additionally, it's weird to me that Gists live under a separate UI instead of the unified GitHub UI. 

In a perfect world, Gists would have search, organization, and be in the same UI as repos and users are instead of needing to "go back to GitHub".

Also, Gists aren't even a category (yet are a top-level product of your platform) for this discussion forum, so I thought this was the next best place for it.

**Comment by monperrus** [comment]
> There is no way to search through my various pages of Gists

This can be done with ` user:Justintime50` as additional search expression, eg <https://gist.github.com/search?q=user%3AJustintime50>

Some documentation at: https://gist.github.com/santisbon/2e1e9c69b58bdf4c9624219a44d40d83

**Comment by heaths** [comment]
While I seem to recall gists being searchable for quite sometime, at least as of 2024-10-15 they are. But this could still be improved:

I just added [filtering support](https://github.com/cli/cli/pull/9728) to `gh gist list` because I didn't know [qualifiers existed](https://docs.github.com/en/search-github/searching-on-github/searching-gists) for gists (and there's no REST or GraphQL API I can find for programmatic use).

It's easy to see why someone wouldn't know they exist: compare the top search bar for normal github.com repos:

<img width="1068" alt="image" src="https://github.com/user-attachments/assets/f8bc2eba-72df-45aa-a148-fbd55b5dab84">

To searching gists:

![image](https://github.com/user-attachments/assets/c88865e6-4226-4b9c-a43e-e3deed0f2da7)

(The text input box had focus but no window was shown.)

At the very least it would be great to see a popup window / drop-down menu / whatever that has a link to the qualifiers you can use to search gists.

**Comment by mattheworiordan** [comment]
Hit the same wall. 500 gists over the years and no way to find anything. GitHub search doesn't index gists, and `gh gist list --include-content --filter` is ~30s on my account, because it's serial API calls.

I scratched it: https://github.com/mattheworiordan/gistgrep -> Local mirror, incremental sync, fzf TUI with ranked previews, and on-device AI summaries via Apple Intelligence.

*Search is now sub-100ms once synced*

<img width="1065" height="425" alt="Image" src="https://github.com/user-attachments/assets/7b56257a-b4c2-461c-8615-901b54421b32" />

`brew install mattheworiordan/tap/gistgrep`. macOS only right now (the summary bit uses FoundationModels). Happy to add a Linux port via Ollama if there's interest, please do raise an issue. I hope my little project helps!
